src/test/resources/edu/hm/hafner/coverage/parser/go/go-coverage-atomic.out

```diff
@@ -0,0 +1,12 @@
+mode: atomic
+github.com/example/project/pkg/utils/file1.go:5.12,8.22 3 1
```

Comment:
The format is:
`<FILE>`:`<start row>`.`<start column>`,`<end row>`.`<end column>` `<# of statements>` `<# of statements covered>`

Reply:
Ok, then this format actually provides only line and instruction (or in your case statement) coverage. That means the counters for the line and branch coverage are either 1/1 or 0/1 (covered/total), see next comment. 

The exact numbers of statements can be used to set the instruction coverage. Since we have only files as elements this coverage needs to be put per file.

Reply:
My mistake, the format is actually:
`<FILE>`:`<start row>`.`<start column>`,`<end row>`.`<end column>` `<# of statements covered>` `<# of statements missed>`
And so that's we have a running tallies of the number of covered & missed statements, then resolve everything at the end.

Reply:
This does not make sense when I look at the test case file? All of your lines actually show a coverage > 0. Or am I missing something?

```
mode: atomic
github.com/example/project/pkg/utils/file1.go:5.12,8.22 3 1
github.com/example/project/pkg/utils/file1.go:10.9,15.23 5 3
github.com/example/project/pkg/utils/file1.go:19.16,20.23 2 0
github.com/example/project/pkg/utils/file1.go:24.3,25.25 1 1
github.com/example/project/pkg/utils/file1.go:30.16,32.25 1 1
github.com/example/project/pkg/db/file2.go:12.7,15.7 1 0
github.com/example/project/pkg/db/file2.go:17.7,22.14 3 2
github.com/example/project/cmd/file3.go:10.12,10.21 1 0
github.com/example/project/cmd/file3.go:15.17,17.27 3 0
github.com/example/project/pkg/test/file4.go:10.12,11.21 1 0
github.com/example/project/pkg/test/file4.go:10.12,11.21 1 1
````

I rewrote the test so it matches my coding style:
https://github.com/jenkinsci/coverage-model/pull/99/commits/3ae1498b697aeb365fa02d80eac0e1f0a49b497b

I will try to refactor the parser as well. What is important and missing: your parser produces INSTRUCTION coverage. Which implies line coverage.

Reply:
Let's take the example line:
```
github.com/example/project/pkg/utils/file1.go:5.12,8.22 3 1
```
It'd be read as:
For `file1.go` , lines `5`-`8`, there were `3` statements that were covered and `1` statement that was missed.

Reply:
But how can then be 
report:
```
github.com/example/project/pkg/db/file2.go:12.7,15.7 1 0
github.com/example/project/pkg/db/file2.go:17.7,22.14 3 2
```
actual test:
````
 two -> assertThat(two).hasName("file2.go")
                        .hasMissedLines(12, 13, 14, 15)
                        .hasCoveredLines(17, 18, 19, 20, 21, 22)
```
From the counters I would expect: Lines 12-15 have one statement covered and 0 missed. So we should have:

````
 two -> assertThat(two).hasName("file2.go")
                        .hasCoveredLines(12, 13, 14, 15, 17, 18, 19, 20, 21, 22)
```

Reply:
Do you need some help to get this parser finished? I think we are currently struggling at the missed/covered computation, see comment above.